gapic/src/main/com/google/gapid/util/Logging.java

```diff
@@ -222,30 +235,63 @@ public void close() {
   }
 
   /**
-   * {@link Handler} that keeps a number of messages in memory for later retrieval.
+   * MessageIterator is an iterator for the {@link Buffer}.
+   * As the {@link Buffer} is a fixed size ring buffer, messages can be discarded if the buffer
+   * fills more quickly than it is consumed. This iterator has been implemented to read the next
+   * most recently available messages. Some messages may be skipped if the iterator lags behind
+   * new messages.
+   */
+  private static class MessageIterator implements Iterator<Log.Message> {
```

Comment:
This class breaks the Iterator contract. I would simply not implement Iterator.

Also, you are accessing private member variables of another class, which has the same problems as accessing private methods.